### PR TITLE
Converting Imagenette and Imagewoof datasets to Epochs

### PR DESCRIPTION
**Pull request by BradLarson**
As part of migrating off of the old Batcher API and onto Epochs, this converts the Imagenette and Imagewoof datasets over to the new API. The three examples that used these datasets have also been converted.

A Jupyter notebook that demonstrated the use of Batcher for setting up Imagenette has been removed, as it is no longer relevant with the new API.

This addresses two items in issue #592.

### CODE REVIEW

File: Examples/VGG-Imagewoof/main.swift [hunk 1]
```diff
@@ -24,15 +25,15 @@ let optimizer = SGD(for: model, learningRate: 0.02, momentum: 0.9, decay: 0.0005
 
 print("Starting training...")
 
-for epoch in 1...90 {
+for (epoch, epochBatches) in dataset.training.prefix(epochCount).enumerated() {
```
Comment (8bitmp3) [inline Examples/VGG-Imagewoof/main.swift:28]: @BradLarson 👍 on migration to Epochs. In case I missed it in the discussions here and during the design meetings, does Epoch lib affect training ⏳ significantly vs Batcher? Thanks

Comment (BradLarson) [reply Examples/VGG-Imagewoof/main.swift:28]: @8bitmp3 - I don't believe I've seen a noticeable performance difference, Epochs and Batcher largely perform the same operations under the hood. The primary benefit is a cleaner, more extensible interface that went through a lot of review before migrating up to swift-apis. For example, in these datasets it's easier to see that batches are lazily constructed from files on disk by loading from their URL.

It does become a lot easier to move the latter parts of the processing operations onto X10, where we see some benefits for having normalization, etc. be performed as part of the X10 trace.

This also made it possible to do on-the-fly synthetic data generation for use in benchmarks, which impacted performance there (as opposed to generating a block of random data upfront and shuffling that).

Comment (8bitmp3) [reply Examples/VGG-Imagewoof/main.swift:28]: Thanks for your detailed answer @BradLarson. Cleaner interface is 👍 especially if users want to look under the hood (which is all Swift).

Fyi, I'm putting together a tutorial, and I received a strange little error which may only apply to my setup in Colab. I'm testing VGG right now with Epochs and Imagewoof after your last merge: 
- Using "master" branch, freshly updated for `Imagewoof` in `Datasets` and the training loop in [`VGG.swift`](https://github.com/tensorflow/swift-models/blob/master/Models/ImageClassification/VGG.swift ).
- I manually defined the `VGGBlock` struct (`Layer` protocol), then ran the cell. 
- Used `VGGBlock` to manually define the `VGG16` struct to set up the neural net architecture. Ran the cell.
- ```
  ...error: type of expression is ambiguous without more context
          let backbone = input.sequenced(through: layer1, layer2, layer3, layer4, layer5)
  ```
- Highlighting it here because yesterday I ran the old code with the `tensorflow-0.9` branch and it was OK. 
- This doesn't happen and all runs smooths if I define both the `VGGBlock` and `VGG16` structs in one cell.
- There is a chance I have a bug that's only applicable to my setup 🙂.
